.github/workflows/beta_publish.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish Beta Release
+
+on:
+  workflow_dispatch:
+    branches:
+      - beta/*
+
+jobs:
+  publish-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14'
+          check-latest: true
+      - name: npm install
+        run: npm install
+      - name: npm version and publish
+        run: |
+          # Example GITHUB_REF for this action would be refs/heads/beta/0.11.0-beta.1 so ${GITHUB_REF:16} resolves to 0.11.0-beta.1
+          export BETA_VERSION=${GITHUB_REF:16}
+          echo "BETA_VERSION=${BETA_VERSION}"
+
+          # Test that the branch name matches the required pattern for beta versions, i.e. 0.11.0-beta.1
+          (echo "$BETA_VERSION" | grep -Eq "^[0-9]+\.[0-9]+\.[0-9]+-beta\.[0-9]+") && echo "Branch name compliant" || (echo "Branch name did not match required pattern"; exit 1)
```

Comment:
I tested this on my local command line and from a command line from an Ubuntu image